src/main.rs

```diff
@@ -1,87 +1,75 @@
+    .version(crate_version!())
     .author("Caleb Meredith <calebmeredith8@gmail.com>\nVictor M. Suarez <svmnotn@gmail.com>")
-    .about("Accelerate back and forth through time for your database or other in-place systems")
-    .args_from_usage(
-     "<url>                 'the targeted url to accelerate'
-      --directory=[path] -d 'the directory holding the motions (defaults to the current dir)'
-      --yes              -y 'auto accept the changes, used for subcommands that remove information (USE WITH CAUTION!)'")
-    .subcommand(SubCommand::with_name("redo").about("subtracts then adds the last motion"))
-    .subcommand(SubCommand::with_name("up").about("adds all remaining motions"))
-    .subcommand(SubCommand::with_name("down").about("subtracts all previous motions"))
-    .subcommand(SubCommand::with_name("reset").about("subtracts then adds all previous motions"))
-    .subcommand(
-      SubCommand::with_name("ls")
-      .about("lists all motions in the directory")
-      .arg_from_usage("--long -l 'Display all motion information'")
-    )
-    .subcommand(
-      SubCommand::with_name("create")
-      .about("create a new motion using the template")
-      .arg_from_usage("<name> 'the name to use for the new motion'")
-    )
-    .subcommand(
-      SubCommand::with_name("add")
-      .about("adds n motions")
-      .arg_from_usage("[n] 'how many motions to add, defaults to 1'")
-    )
-    .subcommand(
-      SubCommand::with_name("sub")
-      .about("subtracts n motions")
-      .arg_from_usage("[n] 'how many motions to substract, defaults to 1'")
-    )
-    .subcommand(
-      SubCommand::with_name("shift")
-      .about("goes to the nth motion relative to the current motion")
-      .arg_from_usage("--n=<n> 'the amount of motions to move relative to the current one, must be given as a number in the form of --n=[number]'")
-    )
-    .subcommand(
-      SubCommand::with_name("goto")
-      .about("go to the nth motion")
-      .arg_from_usage("<n> 'the motion to go to, 0 based'")
-    )
-    .get_matches();
+    .about("Accelerate back and forth for your database schema or other in-place systems")
+    .settings(&[
+      AppSettings::GlobalVersion,
+      AppSettings::VersionlessSubcommands,
+      AppSettings::UnifiedHelpMessage,
+      AppSettings::ArgRequiredElseHelp
+    ])
+    .args(&[
+      Arg::with_name("url").takes_value(true).required(true).help("The url to the database which will be accelerated"),
+      Arg::with_name("directory").long("directory").short("d").takes_value(true).value_name("PATH").help("The directory holding the motions (defaults to the current dir)"),
+      Arg::with_name("yes").long("yes").short("y").help("Bypass the confirmation step for running subtraction motions")
+    ])
+    .subcommands(vec![
+      SubCommand::with_name("ls").about("Lists all motions in the directory").arg(Arg::with_name("verbose").short("v").help("Display all of the motions information.")),
```

Comment:
There are a couple things I changed, what specifically? 😊

I like the `.xxx()` syntax because it is more explicit/detailed and I like the arrays because it indents a bit better than a bunch of `.subcommand`s or `.arg`s.